src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -0,0 +1,374 @@
+            customer_container_template["ingress"] = ingress_def
+            customer_container_template["registryCredentials"] = registry_def
+
+        self.session_pool_def["properties"]['customContainerTemplate'] = customer_container_template
+        self.session_pool_def["properties"]["secrets"] = secrets_def
+        self.session_pool_def["properties"]["dynamicPoolConfiguration"] = dynamic_pool_def
+        self.session_pool_def["properties"]["sessionNetworkConfiguration"] = session_network_def
+        self.session_pool_def["properties"]["scaleConfiguration"] = session_scale_def
+
+    def create(self):
+        try:
+            return self.client.create(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                name=self.get_argument_name(),
+                session_pool_envelope=self.session_pool_def, no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+
+class SessionPoolUpdateDecorator(SessionPoolPreviewDecorator):
+    def update(self):
+        try:
+            return self.client.update(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                name=self.get_argument_name(),
+                session_pool_envelope=self.session_pool_def, no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def construct_payload(self):
+        self.session_pool_def["location"] = self.get_argument_location()
+
+        # Take None or Value, validation is in RP side.
+        self.session_pool_def["properties"]["containerType"] = self.get_argument_container_type()
+        self.session_pool_def["properties"]["environmentId"] = self.get_argument_managed_env()
+
+        # DynamicPoolConfiguration
+        dynamic_pool_def = None
+        if self.get_argument_cooldown_period_in_seconds() is not None:
+            dynamic_pool_def = {}
+            dynamic_pool_def["cooldownPeriodInSeconds"] = self.get_argument_cooldown_period_in_seconds()
+
+        # SessionNetworkConfiguration
+        session_network_def = None
+        if self.get_argument_egress_enabled() is None:
+            session_network_def = {}
+            session_network_def['status'] = convert_egress_parameter(self.get_argument_egress_enabled())
+
+        # ScaleConfiguration
+        session_scale_def = None
+        if self.get_argument_max_concurrent_sessions() is not None or self.get_argument_ready_session_instances() is not None:
+            session_scale_def = {}
+            session_scale_def["maxConcurrentSessions"] = self.get_argument_max_concurrent_sessions()
+            session_scale_def["readySessionInstances"] = self.get_argument_ready_session_instances()
+
+        secrets_def = None
+        if self.get_argument_secrets() is not None:
+            secrets_def = parse_secret_flags(self.get_argument_secrets())
+
+        # CustomerContainerTemplate
```

Comment:
Base on different part recommend to use `set_up_xxxx()` to recognize the logic, for example:
```
    def construct_payload(self):
        self.set_up_dynamic_configuration()
        self.set_up_scale_configuration()
        self.set_up_customer_container_template()
        ...
```